src/main/java/teho/main/Parser.java

```diff
@@ -0,0 +1,44 @@
+package teho.main;
+
+import teho.main.Deadline;
+import teho.main.Event;
+import teho.main.Task;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+/**
+ * Makes sense of the user command that was loaded in file.
+ */
+public class Parser {
+    /**
+     * Parses input and adds task to taskList.
+     *
+     * @param nextLine Line to be parsed.
+     * @param taskList List to add the parsed task into.
+     */
+    public static void readLine(String nextLine, ArrayList<Task> taskList) {
+        Task taskToAddFirst;
+        //use (\\|) to treat \ as a legit character or else it will get split by any spaces
+        String[] sections = nextLine.split("\\|");
+        int taskNumber = taskList.size(); //counting from 0
+        if (sections[0].equals("T")) {
+            taskToAddFirst = new ToDo(sections[2]);
+            taskList.add(taskToAddFirst);
+            if (sections[1].equals("1")) {
```

Comment:
You can simplify your code by creating the task first and then checking if it should be marked as done. This eliminates code duplication and makes it easier to read. (i.e repeating the if sections[1] == 1 check)